src/main/java/org/folio/linked/data/service/resource/ResourceMarcServiceImpl.java

```diff
@@ -230,8 +245,43 @@ private boolean isMonograph(String leader) {
     return isLanguageMaterial(leader.charAt(6)) && isMonographicComponentPartOrItem(leader.charAt(7));
   }
 
+  private Optional<ResponseEntity<Record>> getRecord(String inventoryId) {
+    try {
+      return Optional.of(srsClient.getFormattedSourceStorageInstanceRecordById(inventoryId));
+    } catch (FeignException.NotFound e) {
+      return Optional.empty();
+    }
+  }
+
+  private Optional<org.folio.ld.dictionary.model.Resource> getResource(String inventoryId) {
+    return getRecord(inventoryId)
+      .map(HttpEntity::getBody)
+      .map(Record::getParsedRecord)
+      .map(ParsedRecord::getContent)
+      .map(this::toJsonString)
+      .flatMap(marcBib2ldMapper::fromMarcJson);
+  }
+
+  private Long save(org.folio.ld.dictionary.model.Resource modelResource) {
+    var id = modelResource.getId();
+    if (resourceRepo.existsById(id)) {
+      throw new AlreadyExistsException(format("Another resource with ID %s already exists in the graph", id));
+    }
+
+    var srsId = modelResource.getFolioMetadata().getSrsId();
+    if (folioMetadataRepository.existsBySrsId(srsId)) {
+      throw new AlreadyExistsException(format("MARC record having srsID %s is already imported to data graph", srsId));
+    }
+
+    return saveAndPublishEvent(resourceModelMapper.toEntity(modelResource), ResourceUpdatedEvent::new);
```

Comment:
Hi @askhat-abishev,
Can we add a comment here why we are emitting ResourceUpdatedEvent?

Emitting ResourceCreatedEvent here will result in sending a "CREATE_INSTANCE" event to Inventory and hence will result in creating a duplicate event in inventory. By emitting ResourceUpdatedEvent, we will send an "UPDATE_INSTANCE" event to inventory there by switching the source of the record as "LINKED_DATA" in inventory.